exchange/src/exchange/rabbitmq/publisher/RabbitMQPublisher.cpp

```diff
@@ -87,20 +80,20 @@ RabbitMQPublisher::broadcast_account_update(
     const std::string& buyer_id = match.buyer_id;
     const std::string& seller_id = match.seller_id;
 
-    auto send_message = [&](const std::string& trader_id, messages::SIDE side) {
-        if (trader_id == "SIMULATED")
+    auto send_message = [&](const std::unique_ptr<manager::generic_trader_t>& trader,
+                            messages::SIDE side) {
+        if (trader->get_type() == manager::BOT)
             return;
         messages::AccountUpdate update = {
-            match.ticker, side, match.price, match.quantity,
-            clients.get_capital(trader_id)
+            match.ticker, side, match.price, match.quantity, trader->get_capital()
         };
         std::string buffer;
         glz::write<glz::opts{}>(update, buffer);
-        publish_message(trader_id, buffer);
+        publish_message(trader->get_id(), buffer);
     };
 
-    send_message(buyer_id, messages::SIDE::BUY);
-    send_message(seller_id, messages::SIDE::SELL);
+    send_message(clients.get_trader(buyer_id), messages::SIDE::BUY);
+    send_message(clients.get_trader(seller_id), messages::SIDE::SELL);
```

Comment:
imo the match should store a ptr to the buyer and seller instead of their string ids so you don't have to do the `O(n)` hashing each time